### PR TITLE
ColorBox: text input should be empty when value is undefined and editAlphaChannel is true (T1115876)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js [hunk 1]
```diff
@@ -578,6 +578,22 @@ QUnit.module('Color Box', {
...
+        { value: null, editAlphaChannel: false },
+        { value: null, editAlphaChannel: true }
+    ].forEach(({ value, editAlphaChannel }) => {
+        QUnit.test(`Text should be empty when value=${value} and editAlphaChannel=${editAlphaChannel}`, function(assert) {
+            const $colorBox = $('#color-box').dxColorBox({
+                value,
+                editAlphaChannel
+            }); const colorBox = $colorBox.dxColorBox('instance');
+
+            assert.notOk(colorBox.option('text'));
```
Comment (EugeniyKiyashko) [inline testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js]: Will be better to do `strictEqual` for this expected value. It is not clear to me that means notOk - '', 0, 'undefined', null

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js [hunk 2]
```diff
@@ -578,6 +578,22 @@ QUnit.module('Color Box', {
...
+        { value: undefined, editAlphaChannel: false },
+        { value: undefined, editAlphaChannel: true },
+        { value: null, editAlphaChannel: false },
+        { value: null, editAlphaChannel: true }
+    ].forEach(({ value, editAlphaChannel }) => {
+        QUnit.test(`Text should be empty when value=${value} and editAlphaChannel=${editAlphaChannel}`, function(assert) {
+            const $colorBox = $('#color-box').dxColorBox({
+                value,
+                editAlphaChannel
+            }); const colorBox = $colorBox.dxColorBox('instance');
```
Comment (EugeniyKiyashko) [inline testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js]: Break line?

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js [hunk 3]
```diff
@@ -578,6 +578,22 @@ QUnit.module('Color Box', {
         colorBox.open();
         assert.equal(colorBox._colorView.option('stylingMode'), 'underlined');
     });
+
+    [
+        { value: undefined, editAlphaChannel: false },
```
Comment (EugeniyKiyashko) [inline testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js:583]: Do we need check the '' value and 0 value? I guess that the result should be simular

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js [hunk 4]
```diff
@@ -578,6 +578,22 @@ QUnit.module('Color Box', {
...
+        { value: undefined, editAlphaChannel: true },
+        { value: null, editAlphaChannel: false },
+        { value: null, editAlphaChannel: true }
+    ].forEach(({ value, editAlphaChannel }) => {
+        QUnit.test(`Text should be empty when value=${value} and editAlphaChannel=${editAlphaChannel}`, function(assert) {
+            const $colorBox = $('#color-box').dxColorBox({
```
Comment (EugeniyKiyashko) [inline testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js:600]: Let's get the instance here without get the root element

File: testing/testcafe/tests/editors/colorbox/colorbox.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+test('Colorbox should display full placeholder', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await t
+    .expect(await takeScreenshot('colorbox-with-placeholder.png', '#container'))
+    .ok()
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
```
Comment (EugeniyKiyashko) [inline testing/testcafe/tests/editors/colorbox/colorbox.ts]: We may return the createWidget() here directly

File: js/ui/color_box/color_box.js [hunk 6]
```diff
@@ -301,7 +301,7 @@ const ColorBox = DropDownEditor.inherit({
 
     _renderValue: function() {
         const value = this.option('value');
```
Comment (EugeniyKiyashko) [inline js/ui/color_box/color_box.js]: const { value, editAlphaChannel } = this.option();
const convertToColor = value && editAlphaChannel;

File: js/ui/color_box/color_box.js [hunk 7]
```diff
@@ -301,7 +301,7 @@ const ColorBox = DropDownEditor.inherit({
...
-        const convertToColor = value !== null && this.option('editAlphaChannel');
+        const convertToColor = value && this.option('editAlphaChannel');
```
Comment (EugeniyKiyashko) [inline js/ui/color_box/color_box.js]: shouldConvertToColor?

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js (hunk 3 above)
Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js:583]: added '' value, digits have another behavior, there are displayed in input even without `editAlphaChannel`, but this logic is not within `_renderValue`

**Review by marker-dao — DISMISSED** [review]

**Review by ksercs — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**